src/mxbase.jl

```diff
@@ -42,24 +42,24 @@ function get_paths()
         end
         default_startcmd = "exec $(Base.shell_escape(default_startcmd)) -nosplash"
     elseif is_windows()
-        default_startcmd = joinpath(matlab_homepath, "bin", (Sys.WORD_SIZE == 32 ? "win32" : "win64"), "MATLAB.exe")
+        default_startcmd = joinpath(matlab_homepath, "bin", (Sys.ARCH === :i686  ? "win32" : "win64"), "MATLAB.exe")
         if !isfile(default_startcmd)
             error("The MATLAB path is invalid. Set the MATLAB_HOME evironmental variable to the MATLAB root.")
         end
         default_startcmd *= " -nosplash"
     end
 
     # Get path to MATLAB libraries
-    if is_linux()
-        matlab_library_dir = Sys.WORD_SIZE == 32 ? "glnx86" : "glnxa64"
+    matlab_library_dir = if is_linux()
+        Sys.ARCH === :i686 ? "glnx86" : "glnxa64"
     elseif is_apple()
-        matlab_library_dir = Sys.WORD_SIZE == 32 ? "maci" : "maci64"
+        Sys.ARCH === :i686 ? "maci" : "maci64"
     elseif is_windows()
-        matlab_library_dir = Sys.WORD_SIZE == 32 ? "win32" : "win64"
+        Sys.ARCH === :i686 ? "win32" : "win64"
```

Comment:
well my os is x86_64 .... I think yuyichao's point was that this depends on how the binaries were built and since I downloaded them of julialang website, that it is expected since the binaries for  32bit julia where probably built on a 32bit system?